File: ruby/testcode/loginjection_016.rb
require_relative 'shared'
require 'logger'

# vuln-code-snippet start ruby_loginj_integer_only
def log_integer_safe(req)
  user_id = req.param('user_id').to_i
  logger = Logger.new($stdout)
  logger.info("Login user_id=#{user_id}") # vuln-code-snippet safe-line ruby_loginj_integer_only
  BenchmarkResponse.ok('ok')
end
# vuln-code-snippet end ruby_loginj_integer_only

File: rust/testcode/loginjection_016.rs
//! CWE-117: User input filtered to printable ASCII range before logging.

// vuln-code-snippet start testcodeLoginjection016
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let user_input = req.param("data");

    let filtered = filter_printable_ascii(&user_input); // vuln-code-snippet target-line testcodeLoginjection016
    eprintln!("[INFO] input={}", filtered);

    super::shared::BenchmarkResponse::ok("Logged")
}

fn filter_printable_ascii(input: &str) -> String {
    input.chars().filter(|c| (*c as u32) >= 0x20 && (*c as u32) <= 0x7E).collect()
}
// vuln-code-snippet end testcodeLoginjection016
